kuma/static/js/components/mathml.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * vanilla version of the function at:
+ * https://github.com/mdn/kuma/blob/master/kuma/static/js/wiki.js#L252
+ */
+(function() {
+    'use strict';
+
+    /*
+     * Bug 981409 - Add some CSS fallback for browsers without MathML support.
+     *
+     * This is based on:
+     * https://developer.mozilla.org/en-US/docs/Web/MathML/Authoring#Fallback_for_Browsers_without_MathML_support
+     * and https://github.com/fred-wang/mathml.css
+     */
+    var mathElements = document.querySelectorAll('math');
+    // if no `math` elements on the page, bail
+    if (!mathElements.length) {
+        return;
+    }
+
+    /**
+     * Returns a `math` element wrapped in a `div` that is positioned offscreen
+     * @returns `div` element
+     */
+    function getMathElement() {
+        var offscreenContainer = document.createElement('div');
+        var mathMLNamespace = 'http://www.w3.org/1998/Math/MathML';
+        var mathElement = document.createElementNS(mathMLNamespace, 'math');
+        var mspaceElement = document.createElementNS(mathMLNamespace, 'mspace');
+
+        mspaceElement.setAttribute('height', '23px');
+        mspaceElement.setAttribute('width', '77px');
+
+        mathElement.append(mspaceElement);
+        offscreenContainer.append(mathElement);
+        offscreenContainer.classList.add('offscreen');
+
+        return offscreenContainer;
+    }
+
+    // Test for MathML support
+    var mathMLTestElement = document.body.appendChild(getMathElement());
+    var box = mathMLTestElement.querySelector('mspace').getBoundingClientRect();
+    document.body.removeChild(mathMLTestElement);
+
+    var supportsMathML =
+        Math.abs(box.height - 23) <= 1 && Math.abs(box.width - 77) <= 1;
+    if (!supportsMathML) {
+        // Add CSS fallback
+        var polyfill = document.createElement('link');
+        polyfill.href = mdn.staticPath + 'styles/libs/mathml.css';
+        polyfill.rel = 'stylesheet';
+        polyfill.type = 'text/css';
+
+        document.head.append(polyfill);
+
+        // if we do not do this, React hydrate is unhappy and removes the element
+        // Warning: Did not expect server HTML to contain a <div> in <div>.
```

Comment:
Why is this code not written in React? Mixing custom DOM manipulation stuff with React is dangerously fragile. 
For example, if we decide to change the React component...
```diff
-id="wikiArticle"
+id="wiki-article"
````
who's going to remember to look up every other .js file to see if it depends on it being called exactly that?

If the only reason for doing it in vanilla JS is to try to hack it so it works in the Wiki AND in the read-only site, then remember that the Wiki is definitely going away. Can we not live without this for another couple of months? Or, fix it in React and just accept previewing in the Wiki is a bit different. 

Sorry for sounding negative but I wonder if this PR just introduces more fragility than it's worth.